src/main/rx/rx.c

```diff
@@ -363,22 +376,37 @@ static uint8_t updateLinkQualitySamples(uint8_t value)
 }
 #endif
 
-static void setLinkQuality(bool validFrame)
+static void setLinkQuality(bool validFrame, timeDelta_t currentDeltaTime)
 {
 #ifdef USE_RX_LINK_QUALITY_INFO
     // calculate new sample mean
     linkQuality = updateLinkQualitySamples(validFrame ? LINK_QUALITY_MAX_VALUE : 0);
 #endif
 
     if (rssiSource == RSSI_SOURCE_FRAME_ERRORS) {
-        setRssi(validFrame ? RSSI_MAX_VALUE : 0, RSSI_SOURCE_FRAME_ERRORS);
+        if (rxConfig()->serialrx_provider == SERIALRX_SBUS) {
```

Comment:
Why is this only applied when frame error based 'RSSI' is used **with SBus**? If this is an effective way of filtering, shouldn't it be used in all cases of frame error based 'RSSI'?

Reply:
@mikeller In practise I have seen the "RSSI" from SBus to fluctuate quite rapidly with the current filtering, sometimes momentarily dipping below osd_rssi_alarm (say 70%) to trigger a warning when the average isn't really very close to that level, hence the desire to improve the filtering.

When I originally implemented this on OpenPilot the filtering algorithm was such that the reported value moved more slowly and was more useful. See https://www.youtube.com/watch?v=CY4v-un00o8 for a demonstration of this. The frame error rate doesn't drop like a rock, leading your model to do the same as you suggest.

When I implemented this functionality on Betaflight, see https://github.com/betaflight/betaflight/pull/5090, I used the existing filtering, in order to minimise change.

This PR originally applied the PT1 filter to all frame based RSSI, but I trimmed this back to just SBus to try to address the concerns being raised. My code now addresses the issue of potentially variable frame rates and so should have broader applicability and the SBus only operation could easily be expanded again.

Reply:
I certainly don't consider this to be "treacherous and almost devoid of value". It works very well for me. Certainly better than no RSSI at all which is what the only other option is when using genuine Futaba receivers which have only SBus and no separate RSSI output. Indeed with FRSky receivers having a separate output I found the PWM output to directly correlate to the SBus dropped frame flag. For Futaba FASST users this is the only option available and I find it invaluable.